### PR TITLE
#185218904 implement Setup redux 

### PR DESCRIPTION
**Pull request by irakevine**
## What does this PR do?

This PR sets up Redux it help us to manage our app's state in a single place and keep changes in our app more predictable and traceable.

## Description of Task to be completed?

- it allows React components to read data from a Redux Store
- dispatch Actions to the Store to update data. 
- Redux helps apps to scale by providing a sensible way to manage state through a unidirectional data flow model.

## How should this be manually tested?

- Clone the repository using `git clone https://github.com/atlp-rwanda/e-comm-team-techsmith-fn.git` 

- Checkout to the branch using `git checkout ch-redux-setup-185218904`

- Install node packages that the application depends on: `npm i`

- Create a `.env` file in the root directory and add all environment variables required to run the application. The required variables are listed in the `.env.example` file available in root directory of the application.

- Start the application development server: `npm start`

## What are the relevant pivotal tracker/Trello stories?

+ [#185218904](https://www.pivotaltracker.com/story/show/185218904)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "version": "0.0.0",
   "type": "module",
   "scripts": {
+    "start": "react-scripts start",
```
Comment (nishimweprince) [inline package.json:7]: You will have to omit react-scripts in a vite project. Please refer to this [link](https://cathalmacdonnacha.com/migrating-from-create-react-app-cra-to-vite) in section 8 for your reference.

**Review by nishimweprince — CHANGES_REQUESTED** [review]
I suggested changes in comments

**Review by jcharles921 — APPROVED** [review]

**Review by irakozejules — APPROVED** [review]

**Review by nishimweprince — APPROVED** [review]

**Review by Dyves1 — APPROVED** [review]
Great

**Review by ReneNtabana — APPROVED** [review]

**Review by frankflej — APPROVED** [review]
Nice work!

**Review by Tparfaite — APPROVED** [review]

**Review by bergad37 — APPROVED** [review]

**Review by bergad37 — APPROVED** [review]
Cool!well done.

**Pull request merged**